shiny/experimental/ui/_valuebox.py

```diff
@@ -0,0 +1,186 @@
+from __future__ import annotations
+
+import numbers
+from typing import Callable, Optional
+
+from htmltools import Tag, TagAttrs, TagChild, css, div
+
+from shiny._typing_extensions import TypeGuard
+
+from ._card import card, card_body
+from ._card_item import CardItem
+from ._css import CssUnit, validate_css_unit
+from ._fill import bind_fill_role
+from ._layout import layout_column_wrap
+
+
+def is_01_scalar(x: object) -> TypeGuard[float]:
+    return isinstance(x, float) and x >= 0.0 and x <= 1.0
+
+
+# It seems to be to use % over fr here since there is no gap on the grid
+def validate_width_unit(x: str | float) -> str:
+    if is_01_scalar(x):
+        scaled_x = 100 * x
+        scaled_not_x = 100 * (1 - x)
+        return f"{scaled_x}% {scaled_not_x}%"
+
+    # TODO: validateCssUnit() should maybe support fr units?
+    # return(paste(x, collapse = " "))
+    return str(x)
+
+
+def validate_height_unit(x: str | float) -> str:
+    if is_01_scalar(x):
+        return f"{100 * x}%"
```

Comment:
Don't magically convert small numbers to percent; if they want percent, they can pass in a string like `"35%".